Update README.md with TorchTune reference

iseeyuan opened the pull request:
We'd estimate communities like to try ExecuTorch with their own tuned models. Add a reference to TorchTune so that users can use all PyTorch native support in the entire flow.

kartikayk commented on examples/models/llama2/README.md:
````diff
@@ -91,6 +91,8 @@ If you want to deploy and run a smaller model for educational purposes. From `ex
     python -m examples.models.llama2.tokenizer.tokenizer -t tokenizer.model -o tokenizer.bin
     ```
 
+If you want to fine tune your model based on a specific dataset, PyTorch team provides a native-Pytorch library, [TorchTune](https://github.com/pytorch/torchtune), for easily authoring, fine-tuning and experimenting with LLMs.
+
````
```suggestion

## (Optional) Finetuning

If you want to finetune your model based on a specific dataset, PyTorch team provides [TorchTune](https://github.com/pytorch/torchtune) - a native-Pytorch library for easily authoring, fine-tuning and experimenting with LLMs.
```

kartikayk replied:
Please feel free to disregard this suggestion.

mergennachin replied:
@kartikayk can you provide an example one-line-ish command to generate a new fine-tuned state_dict pth file?

kartikayk replied:
Oh sure, let me add this in the suggestion

kartikayk replied:
Done

iseeyuan replied:
Thanks @kartikayk ! Will update.

kartikayk commented on examples/models/llama2/README.md:
```diff
@@ -91,6 +91,20 @@ If you want to deploy and run a smaller model for educational purposes. From `ex
+Once you have [TorchTune installed](https://github.com/pytorch/torchtune?tab=readme-ov-file#get-started) you can finetune Llama2 7B model using LoRA on a single GPU, using the following command:
+
+
+tune run lora_finetune_single_device \
+--config llama2/7B_lora_single_device \
+checkpointer.checkpoint_dir=<path_to_checkpoint_folder>  \
+tokenizer.path=<path_to_checkpoint_folder>/tokenizer.model
```
nit: Should wrap this in a code block (sorry suggestions doesnt allow me to do that)

kartikayk commented on examples/models/llama2/README.md:
```diff
@@ -91,6 +91,20 @@ If you want to deploy and run a smaller model for educational purposes. From `ex
+
+If you want to finetune your model based on a specific dataset, PyTorch provides [TorchTune](https://github.com/pytorch/torchtune) - a native-Pytorch library for easily authoring, fine-tuning and experimenting with LLMs.
+
+Once you have [TorchTune installed](https://github.com/pytorch/torchtune?tab=readme-ov-file#get-started) you can finetune Llama2 7B model using LoRA on a single GPU, using the following command:
```
```suggestion
Once you have [TorchTune installed](https://github.com/pytorch/torchtune?tab=readme-ov-file#get-started) you can finetune Llama2 7B model using LoRA on a single GPU, using the following command. This will produce a checkpoint where the LoRA weights are merged with the base model and so the output checkpoint will be in the same format as the original Llama2 model.
```

kartikayk commented on examples/models/llama2/README.md:
```diff
@@ -91,6 +91,20 @@ If you want to deploy and run a smaller model for educational purposes. From `ex
+--config llama2/7B_lora_single_device \
+checkpointer.checkpoint_dir=<path_to_checkpoint_folder>  \
+tokenizer.path=<path_to_checkpoint_folder>/tokenizer.model
+
```
```suggestion
To run full finetuning with Llama2 7B on a single device, you can use the following command.

tune run full_finetune_single_device \
--config llama2/7B_full_single_device \
checkpointer.checkpoint_dir=<path_to_checkpoint_folder> \
tokenizer.path=<path_to_checkpoint_folder>/tokenizer.model
```

iseeyuan committed:
Update README.md with TorchTune reference (#2873)

Summary:
We'd estimate communities like to try ExecuTorch with their own tuned models. Add a reference to TorchTune so that users can use all PyTorch native support in the entire flow.


Reviewed By: mergennachin, cccclai

Differential Revision: D55807346

Pulled By: iseeyuan
